core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -127,6 +129,40 @@ public void testReplaceTxnBuilder() throws Exception {
         .containsEntry("key2", "value2");
   }
 
+  @Test
+  public void testReplaceTxnBuilderDefaultFormatVersion() throws Exception {
+    HadoopCatalog catalog = hadoopCatalog();
+    TableIdentifier tableIdent = TableIdentifier.of("db", "ns1", "ns2", "tbl");
+
+    Transaction createTxn =
+        catalog
+            .buildTable(tableIdent, SCHEMA)
+            .withPartitionSpec(SPEC)
+            .withProperty("key1", "value1")
+            .createOrReplaceTransaction();
+
+    createTxn.newAppend().appendFile(FILE_A).commit();
```

Comment:
nit: but we could refactor all this out since it's all identical except for the version. Not a big deal now, but maybe makes sense since we'll have a "V3" soon enough

Reply:
I refactored all applicable tests to run against both versions.